### ISSUE TITLE
Update axe-core version for Storybook Accessibility panel to ^4.6.3

### ISSUE THREAD

**Issue by majornista** [adobe/react-spectrum#4010]
# 🙋 Feature Request

Update axe-core version for Storybook Accessibility panel to ^4.6.3

## 🤔 Expected Behavior

The WAI-ARIA APG includes a Switch example that does not have `aria-checked` on the `input`: https://www.w3.org/WAI/ARIA/apg/patterns/switch/examples/switch-checkbox/ and the error flagged in the Storybook accessibility panel is not flagged using the Chrome axe DevTools extension.

## 😯 Current Behavior

Testing [Switch Storybook Example](https://reactspectrum.blob.core.windows.net/reactspectrum/afe118fbd0da1feba632788898442a9c0a89a7c0/storybook/index.html?path=/story/switch--default&providerSwitcher-express=false), the Accessibility panel reports the false positive, "Critical
Required ARIA attribute not present: aria-checked" on an `input[type="checkbox"][role="switch"]` that is flagged as an error in [axe-core 3.5](https://dequeuniversity.com/rules/axe/3.5), but is not an error in [axe-core 4.6](https://dequeuniversity.com/rules/axe/4.6). The WAI-ARIA APG includes a Switch example that does not have `aria-checked` on the `input`: https://www.w3.org/WAI/ARIA/apg/patterns/switch/examples/switch-checkbox/ and the error flagged in the Storybook accessibility panel is not flagged using the Chrome axe DevTools extension.

## 💁 Possible Solution

Update axe-core version [^3.0.3](https://github.com/adobe/react-spectrum/blob/6fb59578b7249bc6ec58a4fe49722c1fb543afc2/package.json#L94) to ^4.6.3.

## 🧢 Your Company/Team

Adobe/Accessibility
